GitHub Deployments: link from repository name to configuration page

gcsecsey opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Resolves DOTDASH-674

## Proposed Changes

* link from repository name to management page

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* To make it easy to update the repository connection

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Apply this branch and run Calypso locally or open the Calypso Live link below
* Navigate to /v2/sites/:siteSlug/deployments and click one of the repository names in the list
* Check that you're redirected to the repository configuration page, and it loads the correct details

https://github.com/user-attachments/assets/846154e3-e9b2-45f3-905e-0b5798a55419


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

gcsecsey committed:
Link from repo name to management page

gcsecsey committed:
Merge branch 'trunk' into dotdash-674-deployments-theres-no-direct-link-from-the-individual

epeicher reviewed (approved):
Thanks @gcsecsey, this improves the UX when interacting with deployments! I have tested it, and it works as expected. I can see the links and they navigate as expected. LGTM! :shipit: 

|Before | After|
|-------|------|
| <img width="2900" height="2828" alt="Image" src="https://github.com/user-attachments/assets/d62b9292-9fa8-41a1-8cc4-2503ecf54eb3" />| <img width="2900" height="2828" alt="Image" src="https://github.com/user-attachments/assets/ef02a177-40fe-4364-9c33-e44c8ecdcd8e" />|

ivan-ottinger reviewed (approved):
Nice improvement! The code looks good and works correctly.

One thing I am curious about is whether it would make sense to introduce a similar back button as we have for repositories:

<img width="2384" height="1938" alt="Markup on 2025-10-21 at 16:19:58" src="https://github.com/user-attachments/assets/ec8a9239-4fe9-4463-acd0-2131e9ff9da3" />

gcsecsey committed:
add back to deployments button to configure page

gcsecsey committed:
Merge branch 'dotdash-674-deployments-theres-no-direct-link-from-the-individual' of github.com:Automattic/wp-calypso into dotdash-674-deployments-theres-no-direct-link-from-the-individual

gcsecsey commented:
> Nice improvement! The code looks good and works correctly.
> 
> One thing I am curious about is whether it would make sense to introduce a similar back button as we have for repositories:

Thanks @ivan-ottinger, this is a great idea IMO! 🙌 I moved the back button to a separate component and added it to the manage page as well.

gcsecsey committed:
Merge branch 'trunk' into dotdash-674-deployments-theres-no-direct-link-from-the-individual
